pkg/agent/flowexporter/exporter/exporter_perf_test.go

```diff
@@ -0,0 +1,269 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exporter
+
+import (
+	"flag"
+	"fmt"
+	"math/rand"
+	"net"
+	goruntime "runtime"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"k8s.io/klog"
+
+	"antrea.io/antrea/pkg/agent/flowexporter"
+	"antrea.io/antrea/pkg/agent/flowexporter/connections"
+	connectionstest "antrea.io/antrea/pkg/agent/flowexporter/connections/testing"
+	"antrea.io/antrea/pkg/agent/flowexporter/flowrecords"
+	interfacestoretest "antrea.io/antrea/pkg/agent/interfacestore/testing"
+	proxytest "antrea.io/antrea/pkg/agent/proxy/testing"
+	"antrea.io/antrea/pkg/ipfix"
+	queriertest "antrea.io/antrea/pkg/querier/testing"
+)
+
+const (
+	testPollInterval           = 2 * time.Second
+	testConnectionsCount       = 30000
+	testUpdateConnectionsCount = 20000
+	testNewConnectionsCount    = 4000
+	testDyingConnectionsCount  = 1000
+	testZoneFilter             = uint16(65520)
+	testDuration               = 15 * time.Second
+	testIdleTimeOut            = 5 * time.Second
+	testPerfActiveFlowTimeout  = 2 * time.Second
+	testPerfIdleFlowTimeout    = 1 * time.Second
+)
+
+var count = 0
+var exp *flowExporter
+
+func BenchmarkExport(b *testing.B) {
+	disableLogToStderr()
+	ctrl := gomock.NewController(b)
+	defer ctrl.Finish()
+
+	registry := ipfix.NewIPFIXRegistry()
+	registry.LoadRegistry()
+	stopChan1 := make(chan struct{})
+	stopChan2 := make(chan struct{})
+	listener := startLocalServer(b, stopChan1, stopChan2)
+
+	conns := getConnections(testConnectionsCount)
+	records := flowrecords.NewFlowRecords()
+	mockIfaceStore := interfacestoretest.NewMockInterfaceStore(ctrl)
+	mockConnDumper := connectionstest.NewMockConnTrackDumper(ctrl)
+	mockProxier := proxytest.NewMockProxier(ctrl)
+	npQuerier := queriertest.NewMockAgentNetworkPolicyInfoQuerier(ctrl)
+	connStore := connections.NewConnectionStore(mockConnDumper, records, mockIfaceStore, true, false, mockProxier, npQuerier, testPollInterval)
+	var err error
+	exp, err = NewFlowExporter(connStore, records, listener.String(), listener.Network(), testPerfActiveFlowTimeout, testPerfIdleFlowTimeout, false, true, false, nil, nil, false)
+	if err != nil {
+		b.Errorf("error in starting flow exporter: %v", err)
+	}
+	// Generate connections for Dumpflows, each round with update connections, new connections and dying connections
+	mockConnDumper.EXPECT().DumpFlows(testZoneFilter).Return(conns, testConnectionsCount, nil).Times(1)
+	for i := 0; i < 5; i++ {
```

Comment:
what does 5 mean here? Are we polling only five times during the test?

Reply:
Yes in the previous setting. Changed to testduration/pollinterval to make it clear.